tests/testthat/test-module_teal.R

```diff
@@ -302,6 +302,36 @@ testthat::describe("srv_teal teal_modules", {
     )
   })
 
+  testthat::it("are called only after teal_data_module is resolved", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = teal_data_module(
+          ui = function(id) actionButton("submit", "click me"),
+          server = function(id) {
+            moduleServer(id, function(input, output, session) {
+              eventReactive(input$submit, teal_data(iris = iris))
+            })
+          }
+        ),
```

Comment:
You can just use a reactive directly. 
```suggestion
        data = eventReactive(input$submit, teal_data(iris = iris)),
```